File: ex1.cpp
//g++ ex1.cpp -o ex1 `pkg-config --cflags --libs opencv`

#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>

#include <iostream>

using namespace std;
using namespace cv;
 
int main() {
  
	Mat img;
	int height, width, pixel;

	img = imread("lena.jpg" , CV_LOAD_IMAGE_GRAYSCALE);  

	if(!img.data) {
		cout <<  "Não foi possivel abrir a imagem" << endl ;
		return -1;
	}
	
	imshow("Imagem 1", img);
  
	width = img.cols;
	height = img.rows;
	
	for(int i = 0; i<width; i++){
		for(int j = 0; j<height; j++){
			pixel = (int)img.at<uchar>(i,j);

			// se pixel+100 > 255 entao pixel = 255 senao pixel = pixel+100
			pixel = ((pixel+100) > 255) ? 255 : pixel+100;
			
			cout << pixel << endl;		
			
			img.at<uchar>(i,j) = pixel;

		}	
	}
	
        imwrite("ex1.jpg",img);
       	
	imshow("Imagem 2", img);
	  
	waitKey(0);
	return 0;
}

File: python/ex1.py
import cv2
import numpy as np

img = cv2.imread('/home/fabia/Imagens/lena.png',0)

cv2.imshow('image',img)

#altura e largura
height, width = img.shape[:2]

for i in range(0,width-1):
	for j in range(0,height-1):
		#print img[i,j]
		pixel = img[i,j]
		pixel = pixel + 100
		if pixel > 255:
			pixel = 255

		print pixel

		img[i,j] = pixel  


cv2.imwrite('ex1.png',img)


img2 = cv2.imread('ex1.png')

cv2.imshow('image2',img2)

cv2.waitKey(0)
cv2.destroyAllWindows()




